File: April 7, 2026.js
function palindromeLocator(str) {

  const invertedS = str.split("").reverse().join("");

  if (invertedS !== str) {
    return "none";
  }

  const middleLength = Math.floor(str.length / 2);

  if (str.length % 2 === 0) {
    return str.slice(middleLength - 1, middleLength + 1);
  }

  return s[middleLength]
}

File: April 7, 2026.py
def palindrome_locator(s):

    inverted_s = s[::-1]
    if inverted_s != s:
        return "none"
    
    middle_length = len(s) // 2
    if len(s) % 2 == 0:
        return s[middle_length - 1:middle_length+1]

    return s[middle_length]